dial_opts.go

```diff
@@ -5,56 +5,108 @@ import (
 	"context"
 	"crypto/tls"
 	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
 
 	"github.com/certifi/gocertifi"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 )
 
 const (
-	// SkipVerifyCA is a constant that imrpoves the readability of `WithSystemCerts`
+	// SkipVerifyCA is a constant that improves the readability of functions
+	// with the insecureSkipVerify parameter.
 	SkipVerifyCA = true
 
-	// VerifyCA is a constant that improves the readability of `WithSystemCerts`.
+	// VerifyCA is a constant that improves the readability of functions
+	// with the insecureSkipVerify parameter.
 	VerifyCA = false
 )
 
-// WithSystemCerts is a dial option for requiring TLS with the system
-// certificate pool.
+// WithSystemCerts return a grpc.DialOption for requiring TLS that is
+// authenticated using the system-provided certificate authority chain.
 //
-// This function panics if the system pool cannot be loaded.
-func WithSystemCerts(insecureSkipVerify bool) grpc.DialOption {
+// If one cannot be found, this falls back to using a vendored version of
+// Mozilla's collection of root certificate authorities.
+func WithSystemCerts(insecureSkipVerify bool) (grpc.DialOption, error) {
 	certPool, err := x509.SystemCertPool()
 	if err != nil {
 		// Fall back to Mozilla collection of root CAs.
 		certPool, err = gocertifi.CACerts()
 		if err != nil {
 			// This library promises that this should never occur.
-			panic("gocertifi returned an error: " + err.Error())
+			return nil, fmt.Errorf("gocertifi returned an error: %w", err)
 		}
 	}
 
 	return grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
 		RootCAs:            certPool,
 		InsecureSkipVerify: insecureSkipVerify,
-	}))
+	})), nil
 }
 
-// WithCustomCerts is a dial option for requiring TLS from a specified path.
-// If the path is a directory, all certs are loaded. If it is an individual
-// file only the directly specified cert is loaded.
+func dirContents(dirPath string) ([][]byte, error) {
```

Comment:
I think it might be worth returning a map from filename to the bytes for that file, even if we don't use the keys. This return signature was non-obvious at first glance